### PR TITLE
vitessdriver: switch to go1.8

### PR DESCRIPTION
**Pull request by sougou**
Issue #2681
We still can't get rid of the legacy 1.7 functions because of how
interfaces work. However, I assume that database/sql won't use them
if it finds the newer 'Context' functions in the driver.

Also, go1.9 has allowed us to implement support for custom types.
We still need to exploit this flexibility to add support for
unsigned and other exotic MySQL types.

### CODE REVIEW

**Comment by sougou** [comment]
@arthurnn @vmg please review.

File: go/vt/vitessdriver/driver.go [hunk 1]
```diff
@@ -209,8 +220,7 @@ func (c *conn) Rollback() error {
 }
 
 func (c *conn) Exec(query string, args []driver.Value) (driver.Result, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
-	defer cancel()
+	ctx := context.TODO()
```
Comment (vmg) [inline go/vt/vitessdriver/driver.go]: I think semantically it may make more sense to use `context.Background()` here.

**Comment by vmg** [comment]
LGTM.

> We still can't get rid of the legacy 1.7 functions because of how
interfaces work. However, I assume that database/sql won't use them
if it finds the newer 'Context' functions in the driver.

Correct. But careful! We cannot drop the old interfaces until the driver targets only Go 1.10+, because of a bug that I discovered and that will be fixed in the next Go release: https://github.com/golang/go/issues/21663

File: go/vt/vitessdriver/driver.go (hunk 1 above)
Comment (sougou) [reply go/vt/vitessdriver/driver.go]: That's what I initially started out with. But then, I realized that these functions are deprecated and must be eventually removed, which swayed me towards a TODO.

Comment (vmg) [reply go/vt/vitessdriver/driver.go]: :+1:

**Pull request merged**